keras/src/trainers/compile_utils_test.py

```diff
@@ -520,3 +520,15 @@ def test_structure_mismatch(self):
         ):
             wrong_struc_y_true = [np.array([[1]])]
             compile_loss(wrong_struc_y_true, y_pred)
+
+    def test_y_true_partial_y_pred_span(self):
+        y_pred = [np.random.random((320, 3))] * 3
+        y_true = np.random.random((320, 3))
+
+        compile_loss = CompileLoss(
+            loss=["mse", None, None], output_names=["a", "b", "c"]
+        )
+        # build call
+        compile_loss(y_true, y_pred)
+        # built call
+        compile_loss(y_true, y_pred)
```

Comment:
Please check that `model.evaluate()` is correct (so that we know we're computing MSE for the first input and not something else, like 1 MSE per input...)

e.g.

```python
inputs = layers.Input((2,))
x = layers.Dense(3, kernel_initializer="ones", use_bias=False)(inputs)
model = models.Model(inputs, [x, x, x])
model.compile(loss=["mse", None, None])
model.fit(x=np.ones((320, 2)), y=np.zeros((320, 3)))
out = model.evaluate(x=np.ones((320, 2)), y=np.zeros((320, 3)))
self.assertAllClose(out, 3.862943)
```

Reply:
I added a test that checks this.